lib/net/http/header.rb

```diff
@@ -806,6 +806,107 @@ def set_form_data(params, sep = '&')
 
   alias form_data= set_form_data
 
+  # Stores form data to be included in the body of a +POST+ or +PUT+ request.
+  # When the request is sent, the data is URL-encoded
+  # (see {URI#encode_www_form}[https://docs.ruby-lang.org/en/master/URI.html#method-c-encode_www_form])
+  # and included in the request body.
+  #
+  # Note: There is no method to retrieve the form data directly;
+  # here we define a convenience method to do just that
+  # (its details are unimportant):
+  #
+  #   def get_form(req)
+  #     YAML.unsafe_load(req.to_yaml.lines[1..].join)['body_data']
+  #   end
```

Comment:
I'm not a fan of this. We shouldn't be trying to find hacky ways to get the internal state of the object.

For `application/x-www-form-urlencoded`, we can just let the reader refer to the documentation of `URI.encode_www_form`.

For `multipart/form-data`, ... I'm not too sure what the best way to show examples of that is.

Reply:
Removed.